File: challenge-15/number_type.py
# Python3 program to find whether input number
# is abundant, perfect or deficient
# Input Constraint: 10 <= input_num <= 2000

# Returns the type of number through integer value
def check_num_type(check_num):
	sum_of_divisors = 1

	# Find and add together all proper divisors of
	# 
	for number in range(2, (check_num // 2) + 1):
		if check_num % number == 0:
			sum_of_divisors += number

	# Greater than 0 means abundant
	# Equal to 0 means perfect
	# Less than 0 means deficient
	return (sum_of_divisors - check_num)

# __main__ script

input_num = 0

# Repeatedly input constraint satisfied
while input_num < 10 or input_num > 2000:
	try:
		input_num = int(input('Input: '))
	except ValueError:
		print('Invalid Input')

if check_num_type(input_num) > 0:
	print(f'{input_num} is an abundant number')
elif check_num_type(input_num) == 0:
	print(f'{input_num} is a perfect number')
elif (check_num_type(input_num)) < 0:
	print(f'{input_num} is a deficient number')

File: challenge-15/number_type.c
/*
* C Program to check whether
* a given number is abundant, perfect or deficient
* Input Constrant: 10 <= input_num <= 2000
*/

#include <stdio.h>

// Returns the type of param num through integer value
int check_num_type(int check_num);

// Driver code
int main()
{
	int input_num;

	// Repeteadly input till constraint satisfied
	do
	{
		printf("Input: ");
		scanf("%d", &input_num);
	} while (input_num < 10 || input_num > 2000);

	if (check_num_type(input_num) > 0)
	{
		printf("%d is an abundant number\n", input_num);
	}
	else if (check_num_type(input_num) == 0)
	{
		printf("%d is a perfect number\n", input_num);
	}
	else if (check_num_type(input_num) < 0)
	{
		printf("%d is a deficient number\n", input_num);
	}

	return 0;
}

// Returns the type of param num through integer value
int check_num_type(int check_num)
{
	int sum_of_divisors = 1;

	// Find sum all the divisors of check_num
	for (int number = 2; number <= check_num / 2; number++)
	{
		if (check_num % number == 0)
		{
			sum_of_divisors += number;
		}
	}

	// If abundant, returns greater than 0
	// If perfect, returns 0
	// If deficient, returns less than 0
	return (sum_of_divisors - check_num);
}	